### DISCUSSION TITLE
Vue warning while using custom useFetch

### DISCUSSION THREAD

**Discussion by k0mar12** [nuxt/nuxt#17992] [Questions]
Hey there.

Here is my custom useFetch 
`composables/useSpaFetch.js`

```
import * as cook from 'set-cookie-parser';
import { setCookie } from 'h3';
import { useNuxtApp } from '#app';

export const useSpaFetch = async (path, options) => {
    const { server } = useRuntimeConfig();
    const { ssrContext } = useNuxtApp();

    return useFetch(path, {
        baseURL: server.baseURL,
       ...options,

        //-- Add headers before make request
        async onRequest({ options }) {
            options.headers = {
                'origin': `${ssrContext.req.headers['x-forwarded-proto']}://${ssrContext.req.headers['host']}`,
                'referer': `${ssrContext.req.headers['x-forwarded-proto']}://${ssrContext.req.headers['host']}/`,
                ...useRequestHeaders(['cookie'])
            };
        },

        //-- After response set new headers
        async onResponse({ response }) {
            for (const cookie of cook.parse(cook.splitCookiesString(response.headers.get('set-cookie')))) {
                setCookie(ssrContext.res, cookie.name, cookie.value, cookie);
            }
        }
    })
}
```

And im trying to use it in my custom plugin, which i imported manual in `nuxt.config.ts`

```
import { defineNuxtPlugin } from '#app';

/**
 * Init data
 *
 */
export default defineNuxtPlugin(async () => {

    await Promise.allSettled([
        //-- Get csrf-cookies
        useSpaFetch('/sanctum/csrf-cookie'),
    ]);
});
```

Everythink works fine, but i got warning:

`[Vue warn]: onServerPrefetch is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement.`

How i can fix it? Thank you :)

P.S.: According to the docs, i can use `useFetch` in `plugins`

**Comment by k0mar12** [comment]
I belive, that problem in `Promise`

How i need to play with this?

**Reply by k0mar12** [reply]
Aha, something still in [dev case](https://v3.vuejs.org/api/sfc-script-setup.html#top-level-await), exactly in Vue 3, i guess.

**Reply by folamy** [reply]
@k0mar12, if you are using nuxt3's useFetch() remember that it is designed only to work during setup or Lifecycle Hooks!
**Within your pages, components, and plugins you can use useFetch to universally fetch from any URL**
Check this out:
[useFetch](https://v3.nuxtjs.org/docs/usage/data-fetching#usefetch)

**Reply by k0mar12** [reply]
@folamy yep, i got it, thanks
